src/pudl/transform/ferc1.py

```diff
@@ -6475,6 +6422,22 @@ def make_calculation_dimensions_explicit(
             the parental dimensions or the child dimensions.
     """
     logger.info(f"Adding {dimensions=} into calculation component table.")
+    # even though we don't actually get correction records for all of the
+    # factiods, we are still going to force them to exist here so all of the
```

Comment:
Should this be for all factoids, or all calculated factoids? If factoids only ever exist as components I'm not sure why we would expect a correction record for them?

Reply:
this is a very good point. this is part of why i wanted to move this into this function instead of do this in `_core_ferc1__table_dimensions`. because these `_correction` records are only ever generated here & they are only used to as the left df in a left merge down below when defining `calc_comps_w_implied_dims`, if they don't show up `null_dim` (which is made out of the calculation components) they will just be correctly lost in the merge.

but that feels pretty opaque. so I'll add another part of the mask to grab only table dim factoids that show up as parents (== calculated values) in the calculation components

Reply:
hm this is less straightforward than i had thought. because this function should reallly be called `make_xbrl_factoid_dimensions_explicit` because this is being used to make dimension explict for dfs that are not the calc comps table. I think it is possible to only add corrections for calculated values but it will be a little complicated. it seems simpler to keep the existing behavior and just explain the left merge drop here.